Objective-C/CBLDictionary.mm

```diff
@@ -321,27 +390,45 @@ - (NSArray*) allKeys {
 
 
 - (void) setChanged {
+    // Has it own _changedLock instead of using the _lock to avoid deadlock when encoding
+    // the object into fleece and at the same time there are some changes in a child
+    // dictionary or array by another thread. There is one side effect of doing this is that
+    // the changed status could be NO when one thread is trying to get the changed status
+    // but the other thread is modifying the object at the same time.
+    [_changedLock lock];
     if (!_changed) {
```

Comment:
This is technically not thread-safe, because the other methods above use a different lock to control access to `_changed`, so it's possible for two threads to read/write it at the same time by holding different locks.